File: Codepath/Week1-Warmup/python/reverse.py
# Problem: Write a function to reverse a string
def reverse(str):
    chars = list(str)
    s = 0; e = len(chars) - 1
    while s < e:
        chars[e], chars[s] = chars[s], chars[e]
        s += 1
        e -= 1
    return ''.join(chars)

File: Codepath/Week1-Warmup/java/reverse.java
/*
 * Problem: Write a function to reverse a string
*/
boolean reverse(String str) {
  char[] chars = str.toCharArray()
  int s = 0, e = chars.length - 1;
  while (s < e) {
    temp = chars[e]
    chars[e] = chars[s]
    chars[s] = temp
    s++;
    e--;
  }
  return String(chars);
}
